odoo/addons/base/models/res_currency.py

```diff
@@ -169,34 +176,40 @@ def is_zero(self, amount):
         return tools.float_is_zero(amount, precision_rounding=self.rounding)
 
     @api.model
-    def _get_conversion_rate(self, from_currency, to_currency):
-        from_currency = from_currency.with_env(self.env)
-        to_currency = to_currency.with_env(self.env)
-        return to_currency.rate / from_currency.rate
+    def _get_conversion_rate(self, from_currency, to_currency, company, date):
+        currency_rates = (from_currency + to_currency).get_rates(company, date)
+        res = currency_rates.get(to_currency.id) / currency_rates.get(from_currency.id)
+        # print (currency_rates.get(to_currency.id), currency_rates.get(from_currency.id), res)
+        return res
 
-    @api.model
-    def _compute(self, from_currency, to_currency, from_amount, round=True):
-        if (to_currency == from_currency):
-            amount = to_currency.round(from_amount) if round else from_amount
-        else:
-            rate = self._get_conversion_rate(from_currency, to_currency)
-            amount = to_currency.round(from_amount * rate) if round else from_amount * rate
-        return amount
-
-    @api.multi
-    def compute(self, from_amount, to_currency, round=True):
-        """ Convert `from_amount` from currency `self` to `to_currency`. """
+    def _convert_amount(self, from_amount, to_currency, company, date, round=True):
```

Comment:
Sorry for nitpicking about the method name, I actually don't like long method names.
Wouldn't `_convert` be clear enough?  Just asking...  Do we expect to convert something else than an amount?
Otherwise I would suggest `_convert_amount_in_case_you_wonder`.  Nah, just kidding :wink:

Reply:
And please add a docstring.

Reply:
:+1: on making `company` and `date` explicit